docs/cheatsheet/boolean.rst

```diff
@@ -0,0 +1,173 @@
+.. _ref_cheatsheet_boolean:
+
+Boolean Expressions
+===================
+
+Boolean expressions can be tricky sometimes, so here are a handful of
+tips and gotchas.
+
+There's a fundamental difference between how ``{}`` is treated by
+:eql:op:`AND` and :eql:op:`OR` vs :eql:func:`all` and :eql:func:`any`.
+The operators :eql:op:`AND` and :eql:op:`OR` require both operands
+to produce a result, which means that an ``{}`` as one of the
+inputs necessarily produces an ``{}`` as the output:
+
+.. code-block:: edgeql-repl
+
+    db> SELECT false AND <bool>{};
+    {}
+    db> SELECT true OR <bool>{};
+    {}
+
+The functions :eql:func:`all` and :eql:func:`any`, however, produce a
+result for all possible input sets, regardless of the number of
+elements:
+
+.. code-block:: edgeql-repl
+
+    db> SELECT all({false, {}});
+    {false}
+    db> SELECT any({true, {}});
+    {true}
+
+Note that expressions like ``{false, {}}`` are equivalent to
+``{false}`` and so the above are just generalizations of boolean
+operators :eql:op:`AND` and :eql:op:`OR` to a set of 1 element. So the
+result for 1 element is fairly intuitive. However, the results
+produced by these functions for ``{}`` may be surprising (even though
+they are mathematically consistent):
+
+.. code-block:: edgeql-repl
+
+    db> SELECT all(<bool>{});
+    {true}
+    db> SELECT any(<bool>{});
+    {false}
+
+There's no direct analogue to the boolean operator "short-circuiting"
+that's implemented in many other languages because in EdgeQL all parts
+of the expression get evaluated. However, there are expressions that
```

Comment:
```suggestion
that's implemented in many other languages because in EdgeQL the 
order of evaluation of subexpressions is generally not defined. 
However, there are expressions that
```

Reply:
Fixed